wolfssl/tests/ping_pong.rs

```diff
@@ -192,18 +193,21 @@ async fn server<S: SockIO>(sock: S, protocol: Protocol) {
     println!("[Server] Finished");
 }
 
+#[test_case(Protocol::DtlsClientV1_2, Protocol::DtlsServerV1_2)]
+#[test_case(Protocol::DtlsClientV1_3, Protocol::DtlsServerV1_3)]
+// TODO: DTLS1.3 <-> DTLS1.2 is not working yet
```

Comment:
Do we have a link to an upstream bug?

My (second hand) understanding was the client 1.2 <-> server 1.3 was ok and only client 1.3 <-> server 1.2 had issues?

Reply:
No, I don't have an exact link to it. A quick search did not find any.
I tested both and it was panicking.
Wanted to revisit this, after `lightway-core` got fixed first, so added this TODO

Reply:
To understand the behavior further, I have extended the tests for TLS too which fails, if we mix protocols:

```
---- tls::client_1_2_server_1_3 stdout ----
[Client] Connecting...
[Server] Connecting...
thread 'tls::client_1_2_server_1_3' panicked at '[Server] try_negotiate: Fatal(FatalError { what: "record layer version error", code: -326 })', wolfssl/tests/ping_pong.rs:169:47
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- tls::client_1_3_server_1_2 stdout ----
[Client] Connecting...
[Server] Connecting...
thread 'tls::client_1_3_server_1_2' panicked at '[Client] try_negotiate: Fatal(FatalError { what: "malformed buffer input error", code: -328 })', wolfssl/tests/ping_pong.rs:121:47
```

Reply:
I found the issue, `Protocol::::DtlsServerV1_3` aka. `wolfssl_sys::wolfDTLSv1_3_server_method()` does not support backward compatibility, we need to use `Protocol::DtlsServer` to support it.

Updated the PR with testcases for all the combinations.

For these cases, panics are expected since there is no backward compatability
`Protocol::DtlsClientV1_3, Protocol::DtlsServerV1_2 => panics;`
`Protocol::DtlsClientV1_2, Protocol::DtlsServerV1_3 => panics;`


Still DTLS client 1.2/1.3 <-> server 1.2 has some issues similar to lightway-core.

Reply:
Well spotted, I had misunderstood what the versioned methods were about, it makes sense as you describe it.